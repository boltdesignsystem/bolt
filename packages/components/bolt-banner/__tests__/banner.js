import { render, stopServer, renderWC } from '../../../testing/testing-helpers';
import schema from '../banner.schema';
const componentSelector = 'bolt-banner';
const { status, align } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is a banner',
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt banner', () => {
  // With a Web Component
  test(`default`, async () => {
    const results = await render('@bolt-components-banner/banner.twig', {
      ...fixtures.defaultData,
    });

    const { innerHTML, outerHTML } = await renderWC(
      componentSelector,
      results.html,
      page,
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
    await expect(innerHTML).toMatchSnapshot();
    await expect(outerHTML).toMatchSnapshot();
  });

  // Without a Web Component
  test(`default`, async () => {
    const results = await render('@bolt-components-banner/banner.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt banner prop -', () => {
  // Target each of the schema keys with the following pattern
  status.enum.forEach(async option => {
    test(`status: ${option}`, async () => {
      const results = await render('@bolt-components-banner/banner.twig', {
        ...fixtures.defaultData,
        status: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  align.enum.forEach(async option => {
    test(`align: ${option}`, async () => {
      const results = await render('@bolt-components-banner/banner.twig', {
        ...fixtures.defaultData,
        align: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
