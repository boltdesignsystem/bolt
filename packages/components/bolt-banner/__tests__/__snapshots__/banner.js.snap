// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt banner default 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner default 2`] = `
<div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
  <slot>
  </slot>
</div>
`;

exports[`Bolt banner default 3`] = `
<bolt-banner>
  This is a banner
</bolt-banner>
`;

exports[`Bolt banner default 4`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - align: center 1`] = `
<bolt-banner align="center">
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - align: end 1`] = `
<bolt-banner align="end">
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-end">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - align: start 1`] = `
<bolt-banner align="start">
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-start">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - status: error 1`] = `
<bolt-banner status="error">
  <div class="c-bolt-banner c-bolt-banner--status-error c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - status: information 1`] = `
<bolt-banner status="information">
  <div class="c-bolt-banner c-bolt-banner--status-information c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - status: success 1`] = `
<bolt-banner status="success">
  <div class="c-bolt-banner c-bolt-banner--status-success c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`Bolt banner prop - status: warning 1`] = `
<bolt-banner status="warning">
  <div class="c-bolt-banner c-bolt-banner--status-warning c-bolt-banner--align-center">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;
